applications/pelion_client/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+cmake_minimum_required(VERSION 3.13.1)
+
+
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE ZDebug)
+endif()
+
+if (CONFIG_NET_L2_OPENTHREAD)
+  set(OT_CONFIG "${CMAKE_CURRENT_SOURCE_DIR}/configuration/common/ot_config.h")
+endif()
```

Comment:
This won't work, as you don't know if `CONFIG_NET_L2_OPENTHREAD=y` or not.

Reply:
What is the correct way to check this?

Reply:
#if IS_ENABLED(CONFIG_NET_L2_OPENTHREAD)

Reply:
but this is CMakeLists.txt. The code here is actually the same as everywhere else. I think it was working fine. But maybe this option is somehow different... Let's double check.

Reply:
Right. So just if should make the job.

See the documentation:

```
 if(<constant>)
 
    True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number. False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND. Named boolean constants are case-insensitive. If the argument is not one of these specific constants, it is treated as a variable or string and 
the following signature is used.
```

Reply:
But in this case what is wrong here? @tejlmand

Reply:
The problem is that if user specify `CONFIG_NET_L2_OPENTHREAD=y` in a config file, this value will be undefined at this point in time, because `find_package(Zephyr)` has not been executed.

Reply:
right - this is done before the boilerplate is executed.
Why do we do it anyway @rakons ? It seems it is indeed not set so I guess we can safely remove it.

Reply:
I see that there is another build define set `OPENTHREAD_PROJECT_CORE_CONFIG_FILE` that pass info about config file.
I am removing it.

Reply:
done